solutions/usaco-919.mdx

```diff
@@ -54,47 +59,40 @@ int main() {
 import java.util.*;
 import java.io.*;
 
-public class paintbarn {
-	static Kattio io;
-	static {
-		try {
-			io = new Kattio("paintbarn");
-		} catch(IOException e) {}
-	}
-
-	static int N, K, A=0;
-	static int[][] psum;
+public class PaintBarn {
+	private static final int WIDTH = 1000;
 	public static void main(String[] args) throws IOException {
-		N = io.nextInt();
-		K = io.nextInt();
-		psum = new int[1001][1001];
-
-		for (int i = 0; i < N; i++) {
-			int a, b, c, d;
-			a = io.nextInt()+1;
-			b = io.nextInt()+1;
-			c = io.nextInt()+1;
-			d = io.nextInt()+1;
-
-			// Setting up the prefix sums array with all the corners of the given rectangle
-			psum[a][b]++;
-			psum[c][d]++;
-			psum[a][d]--;
-			psum[c][b]--;
+		Kattio io = new Kattio("paintbarn");
+		int rectNum = io.nextInt();
+		int paintReq = io.nextInt();
+		int[][] barn = new int[WIDTH + 1][WIDTH + 1];
+
+		for (int i = 0; i < rectNum; i++) {
+			int start_x = io.nextInt();
+			int start_y = io.nextInt();
+			int end_x = io.nextInt();
+			int end_y = io.nextInt();
+
+			// Set up the prefix sums array with all the corners of the given rectangle
+			barn[start_x][start_y]++;
+			barn[end_x][end_y]++;
+			barn[start_x][end_y]--;
+			barn[end_x][start_y]--;
 		}
 
-		// Running 2D prefix sums on the array
-		for (int i = 1; i < 1001; i++) {
-			for (int j = 1; j < 1001; j++) {
-				psum[i][j] += psum[i][j-1] + psum[i-1][j] - psum[i-1][j-1];
-				if (psum[i][j] == K) {
-					A++;
+		int valid_area = 0;
+		// Run 2D prefix sums on the array
+		for (int x = 0; x <= WIDTH; x++) {
+			for (int y = 0; y <= WIDTH; y++) {
+				if (x > 0) barn[x][y] += barn[x - 1][y];
+				if (y > 0) barn[x][y] += barn[x][y - 1];
+				if (x > 0 && y > 0) barn[x][y] -= barn[x - 1][y - 1];
+				if (barn[x][y] == paintReq) {
+					valid_area++;
```

Comment:
i like the og one better :eyes:

Reply:
why lol

Reply:
why we doing all these conditional statements when we can one line it :eyes:

Reply:
no one likes giant one-liners harry